pullRequestHandler.go

```diff
@@ -123,9 +133,97 @@ func hasUnsigned(req types.PullRequestOuter, client *github.Client) (bool, error
 		}
 	}
 
-	return hasUnsigned, err
+	return hasUnsigned
 }
 
 func isSigned(msg string) bool {
 	return strings.Contains(msg, "Signed-off-by:")
 }
+
+func lintCommits(commits []*github.RepositoryCommit) bool {
+	for _, commit := range commits {
+		if commit.Commit != nil && commit.Commit.Message != nil {
+			if lintCommit(commit.Commit.Message) == false {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func lintCommit(message *string) bool {
+	var valid bool
+
+	if message == nil {
+		return false
+	}
+
+	parts := strings.Split(*message, "\n")
+
+	if len(parts) > 0 {
+		lengthValid := len(parts[0]) <= 50
+		var startsWithUpper bool
+
+		firstCharacter := getFirstCharacter(parts[0])
+		if firstCharacter != nil {
+			startsWithUpper = len(*firstCharacter) > 0 && strings.ToUpper(*firstCharacter) == *firstCharacter
+		}
+
+		valid = lengthValid && startsWithUpper
+	}
+
+	return valid
+}
+
+func getFirstCharacter(msg string) *string {
+	var ret *string
+
+	for _, runeVal := range msg {
+		asStr := string(runeVal)
+		ret = &asStr
+		break
+	}
+
+	return ret
+}
+
+func applyLintingLabel(req types.PullRequestOuter, client *github.Client, issue *github.Issue, lintResult bool) error {
+	labelCaption := "review/commit-message"
+	var actionErr error
+	hasLabel := hasLabelAssigned(labelCaption, issue)
+
+	ctx := context.Background()
+
+	if hasLabel {
+		if lintResult == true {
+			res, assignLabelErr := client.Issues.RemoveLabelForIssue(ctx, req.Repository.Owner.Login, req.Repository.Name, req.PullRequest.Number, labelCaption)
+			if assignLabelErr != nil {
+				actionErr = fmt.Errorf("removeLabel: %s limit: %d, remaining: %d", assignLabelErr, res.Limit, res.Remaining)
+			}
+		}
+	} else {
+		if lintResult == false {
+			_, res, assignLabelErr := client.Issues.AddLabelsToIssue(ctx, req.Repository.Owner.Login, req.Repository.Name, req.PullRequest.Number, []string{labelCaption})
+			if assignLabelErr != nil {
+				actionErr = fmt.Errorf("addLabel: %s limit: %d, remaining: %d", assignLabelErr, res.Limit, res.Remaining)
+			}
+			link := fmt.Sprintf("https://github.com/%s/%s/blob/master/CONTRIBUTING.md", req.Repository.Owner.Login, req.Repository.Name)
+
+			body := `Please check that your commit messages fit within [these guidelines](` + link + `):
```

Comment:
> Please check that your commit messages fit within `[these guidelines](link)`:
>
> * Commit subject should not exceed 50 characters
> * Commit subject should start with an uppercase letter

I think there needs to be a quick link here to make it as easy as possible for everyone to edit their commit history. A direct link to [this page](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) (as in the contributing guide) would be perfect.

Prior Slack discussion [here](https://openfaas.slack.com/archives/C80MVK5FH/p1521405255000071?thread_ts=1521404822.000122&cid=C80MVK5FH).

Thoughts?

Reply:
Yes a link could be good for re-writing history.